libslide/src/bignum/compare.rs

```diff
@@ -0,0 +1,131 @@
+use crate::bignum::Bignum;
+use std::cmp::Ordering;
+
+impl Ord for Bignum {
+    fn cmp(&self, other: &Bignum) -> Ordering {
+        let greater: Ordering;
+        let less: Ordering;
+        if self.is_neg && !other.is_neg {
+            return Ordering::Less;
+        } else if other.is_neg && !self.is_neg {
+            return Ordering::Greater;
+        } else if other.is_neg && self.is_neg {
+            greater = Ordering::Less;
+            less = Ordering::Greater;
+        } else {
+            greater = Ordering::Greater;
+            less = Ordering::Less;
+        }
```

Comment:
Why include these branches?

Reply:
because if its a negative number we need to switch the comparison. because we are comparing the magnitude in all the branches? I know this isn't a great way to handle it but I couldn't really think of anything cleaner. Do you have any general ideas?

Reply:
nvm this is fine